public/index.js

```diff
@@ -0,0 +1,159 @@
+  projectId: 'pulse-box',
+  storageBucket: 'pulse-box.appspot.com',
+  messagingSenderId: '118237267477',
+  appId: '1:118237267477:web:ccee5e3ab7928a8ce4301c',
+  measurementId: 'G-0Q1RSLYZVN',
+};
+
+firebase.initializeApp(firebaseConfig);
+
+function googleSignInButton() {
+  const auth = firebase.auth();
+  let provider = new firebase.auth.GoogleAuthProvider();
+  auth.signInWithPopup(provider).catch(function (error) {
+    if (error.code === 'auth/account-exists-with-different-credential') {
+      let pendingCred = error.credential;
+      let email = error.email;
+      auth.fetchSignInMethodsForEmail(email).then(function (methods) {
+        if (methods[0] === 'password') {
+          let password = promptUserForPassword();
+          auth
+            .signInWithEmailAndPassword(email, password)
+            .then(function (user) {
+              return user.linkWithCredential(pendingCred);
+            })
+            .then(function () {});
+          return;
+        }
+        let provider = getProviderForProviderId(methods[0]);
+        auth.signInWithPopup(provider).then(function (result) {
+          result.user
+            .linkAndRetrieveDataWithCredential(pendingCred)
+            .then(function (usercred) {});
+        });
+      });
+    }
+  });
+}
+
+function signOut(event) {
+  firebase
+    .auth()
+    .signOut()
+    .then(function () {
+      // Sign-out successful.
+    })
+    .catch(function (error) {
+      // An error happened.
+    });
+}
+
+function uploadVideoToFirebase(event) {
+  event.stopPropagation();
+  event.preventDefault();
+  const storage = firebase.storage().ref();
+  let file = event.target.files[0];
+  let metadata = {
+    contentType: file.type,
+  };
+
+  firebase.auth().onAuthStateChanged(function (user) {
+    if (user) {
+      let pulseBox = storage.child('data/' + user.uid);
+      console.log(file);
+      // Find all the prefixes and items.
+      pulseBox
+        .listAll()
+        .then(function (res) {
+          res.prefixes.forEach(function (folderRef) {
+            // All the prefixes under listRef.
+            // You may call listAll() recursively on them.;
+          });
+          res.items.forEach(function (itemRef) {
+            let pulseCurrentStorage = storage.child(itemRef.location.path_);
+            pulseCurrentStorage.delete().then(function () {
+              console.log('Cleaning ' + user.uid + ' storage box.');
+            });
+          });
+        })
+        .catch(function (error) {});
+      storage
+        .child('data/' + user.uid + '/' + file.name)
+        .put(file, metadata)
+        .on('state_changed', function (snapshot) {
+          let progress =
+            (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
+          if (progress > 0) {
+            document.getElementById('progress-bar').classList.remove('hidden');
+          }
+          document.getElementById('capture-button').classList.add('hidden');
```

Comment:
Remove this line of code to allow users to re-upload video without having to sign out.